### PR TITLE
docs: document gotcha for global rules bleeding into components

### LINKED ISSUE #213: Generic class names are causing global styles to be applied in some cases
Specific case: disabled Menu item
The disabled menu item has class `disabled`. This results in some global styles being applied. The global styles are coming from some old analytics plugins (used for generating pivot tables event reports, etc). We are hesitant to change these old plugins and wonder if ui-core can take a different approach to avoid having generic classnames. Here is a screenshot of some applied styles:

image

### PR DESCRIPTION
**Pull request by varl**
Resolves #213.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,3 +68,77 @@ automatically starts.
 It is very important that the commit that lands on **master** follows
 the conventional commit format, since that is what is used to
 automatically determine the next version.
+
+**Never push straight to master, always go through a PR!**
```
Comment (Mohammer5) [inline README.md]: We should be able to lock master so that no one can push directly anyways.
Is there a reason why we're not using this feature?

Comment (varl) [reply README.md]: The @dhis2-bot needs to push to master when it cuts the release, that's why it's not locked. I do think it's locked for non-admins though. This is just there for clarity.

**Comment by varl** [comment]
> You've offered a way to fix the styles that are wrongly applied to other elements. The solution includes a class name that has to be passed down, which I think is not ideal.

That's true.
 
> Let's say we have a component somewhere in the code like this:
> 
> ```js-jsx
> <div>
>     {...}
>     <Component disabled />
> </div>
> ```
> 
> And there are styles that change the style for `.disabled` and `div` which affect the component.
> I think the code should be changed to this:
> 
> ```js-jsx
> <div>
>     {...}
>     <div className="fix-container">
>         <Component disabled />
>     </div>
> </div>
> ```
> 
> ```css
> .fix-container .disabled {
>     /* will undo the overriding of global styles for this component */
>     opacity: 1 !important;
> }
> ```
> 
> Then all styles can be overriden, also for nested elements inside `Component`, which wouldn't be accessible through the `className` prop. Also the code for using the component itself doesn't need to be changed. Removing a container once the bleeding-in styles have been removed should be a but easier, could be done through git as well (even if the implementation of `Component` itself has changed) by stripping out the fixing commit (might be easier to do if there are a lot of changes to prevent styles from bleeding in)

Nice! Excellent workaround for the cascading problem.

**Comment by varl** [comment]
@Mohammer5 updated with your workaround.

**Pull request merged**